webapp/src/components/file_preview/file_preview_modal.tsx

```diff
@@ -0,0 +1,41 @@
+import React, {FC, useCallback} from 'react';
+
+import {FileInfo} from 'mattermost-redux/types/files';
+
+import {useDispatch, useSelector} from 'react-redux';
+
+import {closeFilePreview} from 'actions/preview';
+import {filePreviewModal} from 'selectors';
+
+import FullScreenModal from 'components/file_preview/full_screen_modal';
+import WopiFilePreview from 'components/file_preview/wopi_file_preview';
+import FilePreviewHeader from 'components/file_preview/file_preview_header';
+
+const FilePreviewModal: FC = () => {
+    const dispatch = useDispatch();
+    const visible = useSelector(filePreviewModal)?.visible;
+    const fileInfo: FileInfo = useSelector(filePreviewModal)?.fileInfo;
```

Comment:
Instead of using useSelector twice, shouldn't we use it once and store it in a variable?

Reply:
Done